hammer/library/aws/ecs.py

```diff
@@ -0,0 +1,177 @@
+            # AWS does not support filtering, so get all task definition family details for account
+            response = self.account.client("ecs").list_task_definition_families()
+        except ClientError as err:
+            if err.response['Error']['Code'] in ["AccessDenied", "UnauthorizedOperation"]:
+                logging.error(f"Access denied in {self.account} "
+                              f"(ecs:{err.operation_name})")
+            else:
+                logging.exception(f"Failed to list task definitions in {self.account}")
+            return False
+
+        if "families" in response:
+            for task_definition_name in response["families"]:
+                tags = {}
+                logging_enabled = False
+                external_image = False
+                is_privileged = False
+                container_name = None
+                try:
+                    task_definition = self.account.client("ecs").describe_task_definition(
+                        taskDefinition=task_definition_name
+                    )['taskDefinition']
+                    task_definition_arn = task_definition["taskDefinitionArn"]
+                    if "containerDefinitions" in task_definition:
+                        for container_definition in task_definition['containerDefinitions']:
+                            container_name = container_definition["name"]
+                            if container_definition.get('logConfiguration') is None:
+                                logging_enabled = False
+                            else:
+                                logging_enabled = True
+
+                            container_privileged_details = container_definition.get('privileged')
+                            if container_privileged_details is not None:
+                                if container_definition['privileged']:
+                                    is_privileged = True
+                                else:
+                                    is_privileged = False
+
+                            image = container_definition.get('image')
+                            if image is not None:
+                                if image.split("/")[0].split(".")[-2:] != ['amazonaws', 'com']:
+                                    external_image = True
+                                else:
+                                    external_image = False
+
+                        if "Tags" in task_definition:
+                            tags = task_definition["Tags"]
+                        task_definition_details = ECSTaskDefinitions(account=self.account,
+                                                                     name=task_definition_name,
+                                                                     arn=task_definition_arn,
+                                                                     tags=tags,
+                                                                     container_name=container_name,
```

Comment:
This is tricky. There can be multiple containers in the container definitions. You will actually record the name of the last container in container definitions, even if this container doesn't have any issues.
The same for all flags you set in the code above.
I think we should set logging_enabled=False in case we have at least one container in the definition without enabled logging, the same for other flags. Also, we should either use the name of container with issue or pass all container names that have some issues.